Examples/laser_acceleration/laser_acceleration_PICMI.py

```diff
@@ -1,110 +1,161 @@
 # with their index (using Python indexing conventions) or with their name
 # (if the user provided a name)
 # Set the ionization for the species number 1 (Argon)
 # and place the created electrons into the species number 2 (electron)
-plasma['Argon'].activate_ionization(model="ADK", target_species=plasma['e-'])
-
-"""
-Numerics part - can be in separate file
-"""
-import numpy as np
-from scipy.constants import c
-from plasmacode import picmi
+if picmi.codename != 'pywarpx':
+    plasma['Argon'].activate_field_ionization( model           = "ADK",
+                                               product_species = plasma['e-'])
 
-# Define the grid
-nx = 64
-ny = 64
-nz = 480
-xmin = -30.e-6
-xmax = +30.e-6
-ymin = -30.e-6
-ymax = +30.e-6
-zmin = -56.e-6
-zmax = +12.e-6
-v_window = (0., 0., c)
-
-# Setup the grid ; this may be code dependent
-# Note that code specific arguments use the code name as a prefix.
-if picmi.codename == 'fbpic':
-    nr = nx/2.
-    grid = picmi.CylindricalGrid(
-        nr=nr, rmin=0., rmax=xmax, bc_rmax='reflective',
-        nz=nz, zmin=zmin, zmax=zmax, bc_zmin='open', bc_zmax='open',
-        n_azimuthal_modes=2,
-        moving_window_velocity=v_window)
-elif picmi.codename in ['warp', 'warpx']:
+# --- electron bunch
+beam_dist = picmi.GaussianBunchDistribution(
+            n_physical_particles = bunch_physical_particles,
+            rms_bunch_size       = bunch_rms_size,
+            rms_velocity         = bunch_rms_velocity,
+            centroid_position    = bunch_centroid_position,
+            centroid_velocity    = bunch_centroid_velocity )
+beam = picmi.Species( particle_type        = 'electron',
+                      name                 = 'beam',
+                      initial_distribution = beam_dist)
+
+# Numerics components
+# -------------------
+
+if geometry == '3D':
     grid = picmi.Cartesian3DGrid(
-        nx=nx, xmin=xmin, xmax=xmax, bc_xmin='periodic', bc_xmax='periodic',
-        ny=ny, ymin=ymin, ymax=ymax, bc_ymin='periodic', bc_ymax='periodic',
-        nz=nz, zmin=zmin, zmax=zmax, bc_zmin='open', bc_zmax='open',
-        moving_window_velocity=v_window,
-        warpx_max_grid_size=32, warpx_max_level=0)
-        # "max_grid" and "max_level" are specific arguments for the code warpX
-
-# Setup the electromagnetic solver
-smoother = picmi.BinomialSmoother(n_pass=2, compensation=True)
-solver = picmi.ElectromagneticSolver(grid=grid, cfl=1.0,
+        number_of_cells           = [nx, ny, nz],
+        lower_bound               = [xmin, ymin, zmin],
+        upper_bound               = [xmax, ymax, zmax],
+        lower_boundary_conditions = ['periodic', 'periodic', 'open'],
+        upper_boundary_conditions = ['periodic', 'periodic', 'open'],
+        moving_window_velocity    = moving_window_velocity,
+        warpx_max_grid_size       = 32)
+        # Note that code-specific arguments use the code name as a prefix.
+elif geometry == 'RZ':
+    grid = picmi.CylindricalGrid(
```

Comment:
For consistency, I would use the same way of setting the inputs
number_of_cells = [nx//2, nz]
etc

Reply:
OK.